### DISCUSSION TITLE
Source data for Australia?   BOM?

### DISCUSSION THREAD

**Discussion by jamestas** [Pirate-Weather/pirateweather#361] [Ideas]
Hi

thanks for the great work done here; I'm currently pulling some items that I'm unable to obtain from the Home Assistant - Australian Government Bureau of Meteorology integration  (BOM)

The BOM is used Australia-wide and is an incredibly accurate source for Australian weather and alert conditions. 
It is used by all Emergency services.

Unfortunately, the other services based outside of Australia are very inaccurate.

Is there an opportunity to integrate the Pirate feed to use the BOM source?   
It would be an incredible asset for Australian based users.

https://github.com/bremor/bureau_of_meteorology
Official feed info:
http://www.bom.gov.au/catalogue/data-feeds.shtml

Cheers
James
OZ

**Comment by alexander0042** [comment]
Thanks for using Pirate Weather and the positive comments! The good news here is that this is something I've thought about, and it turns out that the BOM publishes the data in the: [format that I'd need here](http://www.bom.gov.au/nwp/doc/access/NWPData.shtml). Once I get the self hosting stuff set up, adding additional data sources like this is next on the docket!

**Reply by cloneofghosts** [reply]
I saw this earlier and was going to post a link to that page but then I saw that you need to be a registered user. I took a look at the linked page (https://reg.bom.gov.au/other/charges.shtml) and noticed that there's a one time fee to register, an annual fee to access the data plus another annual fee to access the dataset.

After doing some conversions it's approximately $6k/year CAD to access the data which seems pretty pricey.

**Reply by jamestas** [reply]
Hi there @cloneofghosts 
You are referencing BOM's commercial offering for realtime data, which they charge for.   

The private / consumer access is free... as in free beer... FREE.  No sla of course.  ;-)

From their site:
Forecasts, warnings and observations
Forecast, warning and observation products and analysis charts are available on our website and by File Transfer Protocol (FTP). These products are **free** to access and are not for commercial use.


I'm already using their free HA integration, but it's not as nice as Pirate's so hence my question about updating some feed sources.

I had assumed that Pirate could be tweaked to point to some 'private' user feeds from the BOM Australian weather, making my HA access non-commercial and still free.  (i.e.  I'm sourcing the data as a private consumer)

Anyway, I will leave it for @alexander0042 to look at. I'm stoked to hear that it has been on the docket for consideration.
**Sing out if you need a tester!**  I will be watching for sure.

BOM: non-commercial info (as above):  
http://www.bom.gov.au/catalogue/data-feeds.shtml




Cheers
James

**Reply by cloneofghosts** [reply]
> Hi there @cloneofghosts You are referencing BOM's commercial offering for realtime data, which they charge for.

I was replying to @alexander0042's post pointing out that the feeds he linked to are not free.

> The private / consumer access is free... as in free beer... FREE. No sla of course. ;-)
> 
> From their site: Forecasts, warnings and observations Forecast, warning and observation products and analysis charts are available on our website and by File Transfer Protocol (FTP). These products are **free** to access and are not for commercial use.

I'm not 100% if those free feeds are in a format that @alexander0042 needs in order to integrate into the API but I'll let him look at those feeds and he'll let you know what is possible.

**Reply by jamestas** [reply]
Yes, I agree.  Might be a challenge, but a good one  ;-)
